ush/python/pyobsforge/obsdb/obsdb.py

```diff
@@ -90,6 +94,16 @@ def get_valid_files(self,
         results = self.execute_query(query, tuple(params))
         valid_files = []
         for row in results:
-            valid_files.append(row[0])
+            filename = row[0]
+            if check_receipt in ["gdas", "gfs"]:
+                query = "SELECT receipt_time FROM obs_files WHERE filename = ?"
+                receipt_time = self.execute_query(query, (filename,))[0][0]
+                receipt_time = datetime.strptime(receipt_time, "%Y-%m-%d %H:%M:%S.%f")
+                print("receipt time, window end:", receipt_time, window_end)
+                print("Types - receipt_time:", type(receipt_time), "window_end:", type(window_end))
+                if receipt_time <= window_end - timedelta(minutes=minutes_behind_realtime[check_receipt]):
```

Comment:
I still don't understand why you need to do this and cannot query within the receipt window?